app/controllers/recruits_controller.rb

```diff
@@ -50,4 +45,13 @@ def destroy
     def recruit_params
       params.require(:recruit).permit(:date_time, :hour, :country, :city, :title, :content)
     end
+
+    # beforeフィルター
+    def profile_blank_check
+      if [current_user.gender, current_user.origin, current_user.current_country,
+          current_user.language_1, current_user.introduce].any?(&:blank?)
+        flash[:alert] = 'Please fill out your profile'
+        redirect_to edit_users_profile_path(current_user)
```

Comment:
flashやredirect_toはアクション内でした方がいいですね。
```suggestion
      [current_user.gender, current_user.origin, current_user.current_country, current_user.language_1, current_user.introduce].any?(&:blank?)
```

Reply:
さっきRailsチュートリアル見たらこんな感じでflashもredirect_toもbefore_actionののメソッドに定義してたいたんですが、アクション内の方が良いのは何か理由がありますか？

```
    # トークンが期限切れかどうか確認する
    def check_expiration
      if @user.password_reset_expired?
        flash[:danger] = "Password reset has expired."
        redirect_to new_password_reset_url
      end
    end
```

Reply:
> アクション内の方が良いのは何か理由がありますか？

メソッド名がcheckなのにやってることはメッセージ格納とredirect_toなので、メソッド内で[確認]以外の処理をしてます。
メソッドには役割以上のことをさせるべきではない(クラスも同様)ので、メソッド名変えるか、個人的にはtrue/falseを返すメソッドにしておくのがシンプルだと思います。

Reply:
> メソッドには役割以上のことをさせるべきではない

なるほどですね。。そっちの方が汎用性あるし他の人が見たときわかりやすいですね。
コード書いてみました。こんな感じですかね？


```

↓コントローラー
def new
  if profile_blank_check?
    flash[:alert] = 'Please fill out your profile'
    redirect_to edit_users_profile_path(current_user)
  else
    @recruit = Recruit.new
  end
end

def create
  @recruit = current_user.recruits.build(recruit_params)
  if profile_blank_check?
    flash[:alert] = 'Please fill out your profile'
    redirect_to edit_users_profile_path(current_user)
  elsif @recruit.save
    flash[:notice] = 'Recruit created!'
    redirect_to root_path
  else
    render :new
  end
end


↓ヘルパー
def profile_blank_check?
  [current_user.gender, current_user.origin, current_user.current_country,
  current_user.language_1, current_user.introduce].any?(&:blank?)
end

```

Reply:
バッチリです。ありがとうございます。